pallets/manta-sbt/src/lib.rs

```diff
@@ -343,17 +377,28 @@ pub mod pallet {
         #[pallet::call_index(1)]
         #[pallet::weight(<T as pallet::Config>::WeightInfo::reserve_sbt())]
         #[transactional]
-        pub fn reserve_sbt(origin: OriginFor<T>) -> DispatchResult {
+        pub fn reserve_sbt(origin: OriginFor<T>, reservee: Option<T::AccountId>) -> DispatchResult {
             let who = ensure_signed(origin)?;
+            // Use reservee account, if None then use account whom signed transaction
+            let reserve_account = reservee.unwrap_or(who.clone());
+
+            let free_account = FreeReserveAccount::<T>::get();
+            // check if account is allowlist account... if it is can do operation for free
+            if free_account.as_ref() != Some(&who) {
+                // Charges fee to tx caller to reserve AssetIds
+                <T as pallet::Config>::Currency::transfer(
+                    &who,
+                    &Self::account_id(),
+                    T::ReservePrice::get(),
+                    ExistenceRequirement::KeepAlive,
+                )?;
+            }
 
-            // Charges fee to reserve AssetIds
-            <T as pallet::Config>::Currency::transfer(
-                &who,
-                &Self::account_id(),
-                T::ReservePrice::get(),
-                ExistenceRequirement::KeepAlive,
-            )?;
-
+            // ensure account does not have any `AssetId` already reserved
+            ensure!(
+                !ReservedIds::<T>::contains_key(&reserve_account),
+                Error::<T>::AssetIdsAlreadyReserved
+            );
```

Comment:
Maybe we should do this check before the potential state change with the Currency::transfer